libs/wingsdk/src/equality.ts

```diff
@@ -0,0 +1,475 @@
+// taken from node 18 https://github.com/nodejs/node/blob/v18.x/lib/internal/util/comparisons.js
```

Comment:
Since this is an older version, are there bugs here that we are relying on? I understand the newer one doesn't work, but do we know what we're missing out on explicitly?

Reply:
what they fixed was taking into account the reference to an object when comparing it- if there are two objects one is the clone of the other, and we add a key (also an object) of the same name and the same structure and different reference, it will fail 
if the parent objects have the same structure and different references - it will pass
(at least that is the impression I got while working on it)